db/mongo/mongoClient.go

```diff
@@ -0,0 +1,106 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	guuid "github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/abdulmoeid7112/student-backend-portal/config"
+	"github.com/abdulmoeid7112/student-backend-portal/db"
+	"github.com/abdulmoeid7112/student-backend-portal/models"
+)
+
+const (
+	collectionName = "Student"
+)
+
+func init() {
+	db.Register("mongo", NewMongoClient)
+}
+
+type client struct {
+	conn *mongo.Client
+}
+
+func (m client) AddStudent(student *models.Student) (string, error) {
+	if student.ID == "" {
```

Comment:
It's better to 

if if student.ID == "" {
      student.ID = guuid.New().String()
}

instead of nesting... nesting more logic makes things complicated.